test/package_analysis/package_summary_test.dart

```diff
@@ -1,80 +1,67 @@
-import 'dart:convert';
 import 'dart:io';
 
 import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
-import 'package:analyzer/file_system/overlay_file_system.dart';
-import 'package:analyzer/file_system/physical_file_system.dart';
 import 'package:pana/src/package_analysis/common.dart';
+import 'package:pana/src/package_analysis/package_analysis.dart';
 import 'package:pana/src/package_analysis/summary.dart';
 import 'package:path/path.dart' as path;
 import 'package:test/test.dart';
 import 'package:yaml/yaml.dart';
 
-Future<void> main() async {
-  final yamlDir = Directory(
-      path.join(path.current, 'test', 'package_analysis', 'testdata'));
-  final indentedEncoder = const JsonEncoder.withIndent('  ');
-  final configTemplate = json.encode({
-    'configVersion': 2,
-    'packages': [
-      {
-        'name': 'test_package',
-        'rootUri': '../',
-        'packageUri': 'lib/',
-        'languageVersion': '2.12'
-      }
-    ],
-    'generated': '2022-01-01T00:00:00.000000Z',
-    'generator': 'pub',
-    'generatorVersion': '2.17.3'
-  });
+import 'common.dart';
 
-  for (final file in await yamlDir.list().toList()) {
+Future<void> main() async {
+  final yamlDir = Directory(path.join(
+    path.current,
+    'test',
+    'package_analysis',
+    'testdata',
+    'summary',
+  ));
+  await for (final file in yamlDir.list()) {
     final doc = loadYaml(await (file as File).readAsString());
     test(doc['name'], () async {
-      final physicalProvider = PhysicalResourceProvider();
-      final provider = OverlayResourceProvider(physicalProvider);
+      // TODO: can we rely on this path being empty and our testing not conflicting with physical files on disk?
+      final packagePath = path.canonicalize(path.join('test_package'));
```

Comment:
can we rely on this path being empty and our testing not conflicting with physical files on disk?
we do not need to create a unique temporary directory _on disk_ because in this case the tests are run on an in-memory test package
no IO disk operations take place during the test
it is enough to know that we have a path which does not coincide with an existing directory

Reply:
OK